src/pybind/mgr/rgw/module.py

```diff
@@ -146,18 +191,84 @@ def _cmd_rgw_realm_rm_zone_creds(self,
+                                                           endpoints,
+                                                           realm_token)
+            return (retval, 'Zone updated successfully', '')
+        except RGWAMException as e:
+            self.log.error('cmd run exception: (%d) %s' % (e.retcode, e.message))
+            return (e.retcode, e.message, e.stderr)
+
     @CLICommand('rgw zone create', perm='rw')
+    @check_orchestrator()
     def _cmd_rgw_zone_create(self,
-                             realm_token : Optional[str] = None,
-                             zonegroup_name: Optional[str] = None,
                              zone_name: Optional[str] = None,
-                             endpoints: Optional[str] = None,
-                             start_radosgw: Optional[bool] = True):
+                             realm_token: Optional[str] = None,
+                             port: Optional[int] = None,
+                             placement: Optional[str] = None,
+                             start_radosgw: Optional[bool] = True,
+                             inbuf: Optional[str] = None):
         """Bootstrap new rgw zone that syncs with existing zone"""
+        try:
+            if inbuf:
+                rgw_specs = self._parse_rgw_specs(inbuf)
+            elif (zone_name and realm_token):
+                placement_spec = PlacementSpec.from_string(placement) if placement else None
+                rgw_specs = [RGWSpec(rgw_realm_token=realm_token,
+                                     rgw_zone=zone_name,
+                                     rgw_frontend_port=port,
+                                     placement=placement_spec)]
+            else:
+                return HandleCommandResult(retval=-errno.EINVAL, stdout='', stderr='Invalid arguments: -h or --help for usage')
+
+            for rgw_spec in rgw_specs:
+                retval, out, err = RGWAM(self.env).zone_create(rgw_spec, start_radosgw)
+                if retval != 0:
+                    break
+
+        except RGWAMException as e:
+            self.log.error('cmd run exception: (%d) %s' % (e.retcode, e.message))
+            return (e.retcode, e.message, e.stderr)
+
+        return HandleCommandResult(retval=retval, stdout=out, stderr=err)
+
+    @CLICommand('rgw zonegroup create', perm='rw')
+    def _cmd_rgw_zonegroup_create(self,
+                                  realm_token: Optional[str] = None,
+                                  zonegroup_name: Optional[str] = None,
+                                  endpoints: Optional[str] = None,
+                                  zonegroup_is_master: Optional[bool] = True):
+        """Bootstrap new rgw zonegroup"""
 
         try:
-            retval, out, err = RGWAM(self.env).zone_create(realm_token, zonegroup_name,
-                    zone_name, endpoints, start_radosgw)
+            retval, out, err = RGWAM(self.env).zonegroup_create(realm_token,
```

Comment:
The realm name should also be passed into this function. As I mention below a realm pull does not need to happen in this situation. It should be removed.

Reply:
I removed this code as it was not used (and not well tested). We can have it back later if we really need this functions.